test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_event_info.cpp

```diff
@@ -0,0 +1,240 @@
+
+        if (status != CL_COMMAND_COMMAND_BUFFER_KHR)
+        {
+            log_error(
+                "ERROR: Incorrect status returned from clGetEventInfo (%d)\n",
+                status);
+
+            return TEST_FAIL;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    clEventWrapper event;
+    cl_int status;
+};
+
+struct CommandQueue : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "Unable to wait for event");
+
+        error = clGetEventInfo(event, CL_EVENT_COMMAND_QUEUE, sizeof(ret_queue),
+                               &ret_queue, &size);
+        test_error(error, "clGetEventInfo failed");
+
+        if (ret_queue != queue)
+        {
+            log_error("ERROR: Wrong command queue returned by clGetEventInfo");
+            return TEST_FAIL;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    clEventWrapper event;
+    cl_command_queue ret_queue = nullptr;
+    size_t size;
+};
+
+struct Context : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
+        error = clWaitForEvents(1, &event);
+        test_error(error, "clWaitForEvents failed");
+
+        error = clGetEventInfo(event, CL_EVENT_CONTEXT, sizeof(ret_context),
+                               &ret_context, &size);
+        test_error(error, "clGetEventInfo failed");
+
+        if (ret_context != context)
+        {
+            log_error("ERROR: Wrong context returned by clGetEventInfo");
+            return TEST_FAIL;
+        }
+
+        return CL_SUCCESS;
+    }
+
+    clEventWrapper event;
+    cl_context ret_context = nullptr;
+    size_t size;
+};
+
+struct ExecutionStatus : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                          nullptr, &event);
+        test_error(error, "clEnqueueCommandBufferKHR failed");
+
```

Comment:
Could we query the execution status here as well, although  valid values will be all of
`status == CL_QUEUED || status == CL_SUBMITTED || status == CL_RUNNING || status == CL_COMPLETE`